packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -86,6 +88,10 @@ export const createPluginActions: CreatePluginActions = ({
     postToContainer(heightChangeMessage(uid, height))
   }
 
+  const onKeydownEsc = () => {
+    postToContainer(modalChangeMessage({ uid, status: false }))
```

Comment:
Hey @Dawntraoz 🙌 
I was wondering if it would make sense to have a `Promise-like` approach as happens for the other state changes.

I mean, besides calling `postToContainer` directly, maybe calling or implementing similar logic as the one implemented in the [setModalOpen](https://github.com/storyblok/field-plugin/pull/404/files?diff=unified&w=1#diff-e62bdc8d2bfaac8dcd9c0ec5f420b868131bc32f10c4be5743ba3e5e28ebe938R109) which returns a promise. Wdyt? 🤔

Reply:
hhhh
I was trying to remember the whole discussion around it once it was some time ago 🫠, but it was basically to allow calling `createFieldPlugin()` multiple times and get rid of the need to have a global state (`FieldPluginContext` for example) to make sure only one instance was set. 

As described [here](https://github.com/storyblok/field-plugin/pull/260) it also enabled a simpler way to exchange/sync the state between the container and field plugin.

>  the functionality I found closer to what I needed to do was in the onHeightChange

I had the same thought since they both handle UI changes and seemed quite similar. However, I considered using the same approach as the other one because it also changes the state of the field plugin, while `onHeightChange` only adjusts the height without impacting the state and also because it's used this way in the `setModalOpen` so we could have all the place that handles the modal having a similar approach. Wdyt? 🤔

Reply:
I see there's a new discussion on the storyfront side, but let me leave a comment here just for the sake of context sharing.

1.
We used to keep the whole state within the library. It means if we call `createFieldPlugin()` multiple times within the same plugin, then then instances of `createFieldPlugin()` can go out of sync. However, after lots of changes on Storyfront and the field plugin library, now the whole state is managed on the storyfront side. So the field plugin only sends deltas to storyfront, and storyfront delivers the whole state. Now field plugin library doesn't need to manually track the whole state.

2.
I think in this particular situation (1) isn't relevant much. esc key handling is just done by the library, and it's not an "action" that is exposed to users.

For example `setModalOpen` is an action.

```
const plugin = useFieldPlugin();

const result = await plugin.actions.setModalOpen(...);
```

However, the esc handling isn't an action. And it's not delivering any return value directly to user. So I think Alba's implementation should work as-is.

Reply:
Thanks @demetriusfeijoo @eunjae-lee 🫂

Actually this is invaluable knowledge that I love you shared with me, thanks a lot! I was doing things more for intuition than actual facts and this conversation has made me understand what's going on 💜

Btw, about the Storyfront side discussion, can you please check my last comment there and let me know if something is not as I think it is? Thanks in advance 🙏